### PR TITLE
[Tabs] Move actions and reducer for the editor tabs into their own files

### LINKED ISSUE #3279: [Tabs] Decouple tabs functionality from sources
We currently show the current selected source tab as active.

We should show a new tab when you're doing a project text search. And it should be a bit active.

This is a bit challenging because tabs are currently associated with sources and URLs

### PR DESCRIPTION
**Pull request by kenjiO**
Fixes #3279.  This is to finish the work started in PR #5672 as discussed in #6711.

Tasks:
- [x] create a reducers/tabs.js file
- [x]  separate tabs actions
- [x]  wire it all up

### CODE REVIEW

**Comment by kenjiO** [comment]
@jasonLaster  can you let me know how this looks so far?

**Comment by jasonLaster** [comment]
looking good. We can land this once travis is happy

**Comment by kenjiO** [comment]
How do you feel about moving the functions `removeSourceFromTabList`, `removeSourcesFromTabList` and `getNewSelectedSourceId` from `reducers/tab.js` to `actions/tab.js` or `utils/tab.js`?  Since they are not used by the reducer or selectors it makes more sense for me to not have them in actions where they are used or in utils.

File: src/reducers/tabs.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  getSourceByURL,
+  getSourceByUrlInSources
+} from "./sources";
+
+import type { Action } from "../actions/types";
+import type { SourcesMap, SourcesState } from "./sources";
+
+type Tab = string;
+export type TabList = Tab[];
+export type TabsState = Tab[];
```
Comment (kenjiO) [inline src/reducers/tabs.js]: I don't know whether or not to keep both `TabList` and `TabState` or just use one.

**Comment by jasonLaster** [comment]
Either is fine

**Comment by jasonLaster** [comment]
This is great! 

This will really help @AnshulMalik out with persisting better tab state. It will also help me with a similar tab related sourcemap bug!

**Comment by kenjiO** [comment]
I changed it to just use the one type `TabList`.   Can you let me know if you want to move the `removeSourceFromTabList`, `removeSourcesFromTabList` and `getNewSelectedSourceId` functions or make any other changes?

**Comment by jasonLaster** [comment]
Thank you! This is a huge step forward

**Comment by jasonLaster** [comment]
Thanks for all the help over the past month. Feel free to join our slack so we can chat more regularly

https://devtools-html-slack.herokuapp.com/

**Pull request merged**